src/libs/fileDownload/FileUtils.ts

```diff
@@ -156,58 +145,61 @@ function appendTimeToFileName(fileName) {
 
 /**
  * Reads a locally uploaded file
- *
- * @param {String} path - the blob url of the locally uplodaded file
- * @param {String} fileName
- * @param {Function} onSuccess
- * @param {Function} onFailure
- *
- * @returns {Promise}
+ * @param path - the blob url of the locally uploaded file
+ * @param fileName - name of the file to read
  */
-const readFileAsync = (path, fileName, onSuccess, onFailure = () => {}) =>
+const readFileAsync: ReadFileAsync = (path, fileName, onSuccess, onFailure = () => {}) =>
     new Promise((resolve) => {
         if (!path) {
             resolve();
+            onFailure('[FileUtils] Path not specified');
+            return;
         }
-
-        return fetch(path)
+        fetch(path)
             .then((res) => {
                 // For some reason, fetch is "Unable to read uploaded file"
                 // on Android even though the blob is returned, so we'll ignore
                 // in that case
                 if (!res.ok && Platform.OS !== 'android') {
                     throw Error(res.statusText);
                 }
-                return res.blob();
-            })
-            .then((blob) => {
-                const file = new File([blob], cleanFileName(fileName), {type: blob.type});
-                file.source = path;
-                // For some reason, the File object on iOS does not have a uri property
-                // so images aren't uploaded correctly to the backend
-                file.uri = path;
-                onSuccess(file);
+                res.blob()
+                    .then((blob) => {
+                        const file = new File([blob], cleanFileName(fileName));
```

Comment:
```suggestion
                        const file = new File([blob], cleanFileName(fileName), {type: blob.type});
```
@luacmartins here's root cause